pinot-common/src/main/java/org/apache/pinot/common/request/context/LiteralContext.java

```diff
@@ -64,6 +65,15 @@ private static BigDecimal getBigDecimalValue(FieldSpec.DataType type, Object val
 
   @VisibleForTesting
   static Pair<FieldSpec.DataType, Object> inferLiteralDataTypeAndValue(String literal) {
+    // Try to interpret the literal as hex encoded bytes
+    try {
+      if (literal.startsWith(BytesUtils.HEX_STRING_PREFIX)) {
```

Comment:
This does not follow the current convention. Why do we need this? We should cast accordingly based on the function signature

Reply:
added literal types and removed the convention based hack